integrations/visual-website-optimizer/lib/index.js

```diff
@@ -182,6 +183,19 @@ function enqueue(fn) {
   window._vis_opt_queue.push(fn);
 }
 
+/**
+ * Check whether the experiment type is one of allowed types.
+ *
+ * @param {Object} experiment
+ * @return {Boolean}
+ */
+function isValidExperimentType(experiment) {
+  var allowedExperimentTypes = ['VISUAL_AB', 'VISUAL', 'SPLIT_URL', 'SURVEY'];
+  var experimentType = (experiment && experiment.type) || '';
```

Comment:
To make it easier to read, and to check the new setting, let's do:
```javascript
function isValidExperimentType(experiment, options) {
  if (!Boolean(this.options.trackOnlyABExperiments)) {
    return true;
  }
  if (!experiment || !experiment.type) {
    return false;
  }
  return allowedExperimentTypes.indexOf(experimentType.toUpperCase()) !== -1;
}
```
Then we can add the `allowedExperimentsTypes` array at the top of the file, so it's only created once.